lib/nats.js

```diff
@@ -801,9 +817,12 @@ Client.prototype.flush = function(opt_callback) {
 
 Client.prototype.publish = function(subject, msg, opt_reply, opt_callback) {
   if (!msg) { msg = EMPTY; }
+  if (typeof subject !== 'string') {
+    this.emit('error', PUB_BAD_SUBJECT_ERR);
+  }
```

Comment:
ok, so these should only be localized and delivered to the callback. Fair. What if the callback is not being provided?